Clusteredverticalbar.js

```diff
@@ -0,0 +1,994 @@
+                    })
+                    .attr('class', 'bar')
+                    .attr("width", x1.bandwidth())
+                    .attr("height", function (d) {
+                        return plotHeight - y(d[d.measure]);
+                    })
+                    .style('fill', function (d, i) {
+                        return UTIL.getDisplayColor(_measure.indexOf(d.measure), _displayColor);
+                    })
+                    .style('stroke', function (d, i) {
+                        return UTIL.getBorderColor(_measure.indexOf(d.measure), _borderColor);
+                    })
+                    .style('stroke-width', 2)
+                    .on('mouseover', _handleMouseOverFn.call(chart, tooltip, svg, _measureProp))
+                    .on('mousemove', _handleMouseMoveFn.call(chart, tooltip, svg, _measureProp))
+                    .on('mouseout', _handleMouseOutFn.call(chart, tooltip, svg, _measureProp))
+                    .on('click', function (d) {
+                        if ($("#myonoffswitch").prop('checked') == false) {
+                            $('#Modal_' + $(div).attr('id') + ' .measure').val(d.measure);
+                            $('#Modal_' + $(div).attr('id') + ' .threshold').val('');
+                            $('#Modal_' + $(div).attr('id') + ' .measure').attr('disabled', true);;
+                            $('#Modal_' + $(div).attr('id')).modal('toggle');
+                        }
+                        else {
+                            var confirm = d3.select('.confirm')
+                                .style('visibility', 'visible');
+                            var _filter = _Local_data.filter(function (d1) {
+                                return d[_dimension[0]] === d1[_dimension[0]]
+                            })
+                            var rect = d3.select(this);
+                            if (rect.classed('selected')) {
+                                rect.classed('selected', false);
+                                filterData.map(function (val, i) {
+                                    if (val[_dimension[0]] == d[_dimension[0]]) {
+                                        filterData.splice(i, 1)
+                                    }
+                                })
+                            } else {
+                                rect.classed('selected', true);
+                                var isExist = filterData.filter(function (val) {
+                                    if (val[_dimension[0]] == d[_dimension[0]]) {
+                                        return val
+                                    }
+                                })
+                                if (isExist.length == 0) {
+                                    filterData.push(_filter[0]);
+                                }
+                            }
+                        }
+                    })
+            }
+            chart.drawPlot = function (data) {
```

Comment:
Let's not create a public function drawPlot on the chart module. We can create a private function or code block which is not accessible to the outsider